File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/ImageRepository.kt
package com.ssafy.ssabree.core.repository

interface ImageRepository {
    /**
        * url로부터 이미지를 가져온다.
        * 순서: 메모리 캐시 -> 로컬 DB -> 원격 다운로드
        */
    suspend fun load(url: String): Result<ByteArray>
}

File: iOS/ssabree/ssabree/Core/Repository/ImageRepository.swift
import Foundation

/// 이미지 캐싱 Repository
/// 순서: 메모리 캐시 -> 로컬 디스크 -> 원격 다운로드
protocol ImageRepository {
    /// URL로부터 이미지 데이터를 가져온다.
    func load(url: String) async -> Result<Data, Error>
}
